### PR TITLE
nil: init at unstable-2022-09-10

### PR DESCRIPTION
**Pull request by figsoda**
###### Description of changes

nil is a new language server for nix

https://github.com/oxalica/nil

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Icy-Thought** [comment]
Result of `nixpkgs-review pr 190594` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nil</li>
  </ul>
</details>

**Review by Icy-Thought — APPROVED** [review]

File: pkgs/development/tools/nil/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "nil";
+  version = "0.0.0-unstable-2022-09-06";
+
+  src = fetchFromGitHub {
+    owner = "oxalica";
+    repo = pname;
+    rev = "b469b939f1ab2550cddbff1a289043c680969d84";
```
Comment (oxalica) [inline pkgs/development/tools/nil/default.nix:10]: I just created a date tag for this.

https://github.com/oxalica/nil/releases/tag/2022-09-10

File: pkgs/development/tools/nil/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "nil";
+  version = "0.0.0-unstable-2022-09-06";
```
Comment (oxalica) [inline pkgs/development/tools/nil/default.nix:5]: Use `unstable-YYYY-MM-DD`. Not `0.0.0`.

**Review by oxalica — CHANGES_REQUESTED** [review]

**Comment by oxalica** [comment]
Also please add an env var `CFG_DATE = <date>` so that the date tag can be include in the `--version` output.

**Review by oxalica — APPROVED** [review]
LGTM.

---

Result of `nixpkgs-review pr 190594` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nil</li>
  </ul>
</details>

**Pull request merged**